test/Microsoft.ML.CpuMath.UnitTests.netcoreapp/UnitTests.cs

```diff
@@ -11,203 +11,210 @@ namespace Microsoft.ML.CpuMath.UnitTests
+            AlignedArray testSrcVectorAligned1 = new AlignedArray(8, _vectorAlignment);
+            AlignedArray testSrcVectorAligned2 = new AlignedArray(16, _vectorAlignment);
             testSrcVectorAligned1.CopyFrom(testSrcVector1, 0, testSrcVector1.Length);
             testSrcVectorAligned2.CopyFrom(testSrcVector2, 0, testSrcVector2.Length);
 
-            testSrcVectors = new AlignedArray[] { testSrcVectorAligned1, testSrcVectorAligned2 };
+            _testSrcVectors = new AlignedArray[] { testSrcVectorAligned1, testSrcVectorAligned2 };
 
-            // Padded destination vectors whose dimensions are multiples of 4
-            float[] testDstVector1 = new float[4] { 0f, 1f, 2f, 3f };
-            float[] testDstVector2 = new float[8] { 0f, 1f, 2f, 3f, 4f, 5f, 6f, 7f };
+            // Padded destination vectors whose dimensions are multiples of 8
+            float[] testDstVector1 = new float[8] { 0f, 1f, 2f, 3f, 4f, 5f, 6f, 7f };
+            float[] testDstVector2 = new float[16] { 0f, 1f, 2f, 3f, 4f, 5f, 6f, 7f, 8f, 9f, 10f, 11f, 12f, 13f, 14f, 15f };
 
-            AlignedArray testDstVectorAligned1 = new AlignedArray(4, SseCbAlign);
-            AlignedArray testDstVectorAligned2 = new AlignedArray(8, SseCbAlign);
+            AlignedArray testDstVectorAligned1 = new AlignedArray(8, _vectorAlignment);
+            AlignedArray testDstVectorAligned2 = new AlignedArray(16, _vectorAlignment);
             testDstVectorAligned1.CopyFrom(testDstVector1, 0, testDstVector1.Length);
             testDstVectorAligned2.CopyFrom(testDstVector2, 0, testDstVector2.Length);
 
-            testDstVectors = new AlignedArray[] { testDstVectorAligned1, testDstVectorAligned2 };
+            _testDstVectors = new AlignedArray[] { testDstVectorAligned1, testDstVectorAligned2 };
         }
 
         [Theory]
-        [InlineData(0, 0, 0, new float[] { 23.28f, -49.72f, 23.28f, -49.72f })]
-        [InlineData(1, 1, 0, new float[] { 204f, 492f, 780f, 1068f })]
-        [InlineData(1, 0, 1, new float[] { 30f, 70f, 110f, 150f, 190f, 230f, 270f, 310f })]
+        [InlineData(0, 0, 0, new float[] { -416.68f, -416.68f, -416.68f, -416.68f, -416.68f, -416.68f, -416.68f, -416.68f })]
+        [InlineData(1, 1, 0, new float[] { 1496f, 3672f, 5848f, 8024f, 10200f, 12376f, 14552f, 16728f })]
+        [InlineData(1, 0, 1, new float[] { 204f, 492f, 780f, 1068f, 1356f, 1644f, 1932f, 2220f, 2508f, 2796f, 3084f, 3372f, 3660f, 3948f, 4236f, 4524f })]
         public void MatMulATest(int matTest, int srcTest, int dstTest, float[] expected)
         {
-            AlignedArray mat = testMatrices[matTest];
-            AlignedArray src = testSrcVectors[srcTest];
-            AlignedArray dst = testDstVectors[dstTest];
+            AlignedArray mat = _testMatrices[matTest];
+            AlignedArray src = _testSrcVectors[srcTest];
+            AlignedArray dst = _testDstVectors[dstTest];
 
             CpuMathUtils.MatTimesSrc(false, false, mat, src, dst, dst.Size);
```

Comment:
One unfortunate result of our unit testing approach is that we are probably only hitting the AVX code paths, since all our CI machines probably have AVX supported.

@tannergooding, do you have any ideas on how we can run the same tests in CI, with different environment variables to turn AVX and both off?

Reply:
The `.netstandard/UnitTests.cs` would be hitting the SSE code paths, since the `CpuMathUtils` for `.netstandard` is hard-coded to call `SseUtils` methods.  I could use the following example to accept an env var:
https://github.com/dotnet/machinelearning/blob/94401d53b6bf41d8688103a7d5697ee251defd46/test/Microsoft.ML.CpuMath.PerformanceTests/SsePerformanceTests.cs#L37-L61

The new code in `CpuMathUtils` could be:
```
private static string _simdArch = Environment.GetEnvironmentVariable("SIMD_ARCH");
private static bool _useAvx = (_simdArch == null) ? true : (string.Equals(_simdArch, "avx", StringComparison.OrdinalIgnoreCase) ? true : false);
private static bool _useSse = (_simdArch == null) ? true : (string.Equals(_simdArch, "sse", StringComparison.OrdinalIgnoreCase) ? true : false);

In the switching function, do:
if (Avx.IsSupported && _useAvx) {...}
else if (Sse.IsSupported && _useSse) {...}
else { // software fallback }
```

How does it sound?

Reply:
There are already environment variables for turning off Intrinsics in .NET Core. 

For now, I think we could use `COMPlus_EnableAVX=false` to disable AVX, and that way test the SSE code path.  I'm not sure how to turn SSE off as well. If we had that, then we could test the software fallbacks too.

See https://github.com/dotnet/coreclr/issues/19519 for adding more flags support.

Reply:
> and that way test the SSE code path. I'm not sure how to turn SSE off as well. 

We can use `COMPlus_FeatureSIMD=0` to disable SSE.